src/clusterfuzz/_internal/bot/tasks/commands.py

```diff
@@ -339,10 +339,16 @@ def process_command_impl(task_name,
             logs.log('Testcase %d platform (%s) does not match with ours\
                  (%s), exiting' % (testcase.key.id(), testcase_platform_id,
                                    current_platform_id))
+            # If this happens on android, we need to make sure we're not
+            # scheduling this task on a subqueue, since we know the current
+            # platform doesn't match.
+            queue = tasks.generic_android_queue() if environment.is_android(
```

Comment:
This might not work actually... if a task runs on the correct device but not the correct branch of android, we might end up in an infinite loop there. We might need to check whether platform actually matches to correctly select the appropriate queue.

Reply:
Fixed this. The idea is that we want to make sure we're trying to schedule on a different device to schedule on the generic android queue. In theory, we should almost never schedule on the generic queue, except while migrating to using subqueues. Even for generic jobs, once the testcase is pinned to a device, we should start scheduling the next tasks on the subqueue so that we keep scheduling on the correct set of devices.